test/extended/networking/network_diagnostics.go

```diff
@@ -0,0 +1,248 @@
+				if _, ok := node.Labels["node-role.kubernetes.io/master"]; !ok {
+					framework.Logf("Diagnostics pod %s is not scheduled to a master node", pod.Name)
+					return false
+				}
+			}
+			cfg, err := oc.AdminConfigClient().ConfigV1().Networks().Get(ctx, clusterConfig, metav1.GetOptions{})
+			if err != nil {
+				framework.Logf("Error getting cluster config: %v", err)
+				return false
+			}
+			return meta.IsStatusConditionTrue(cfg.Status.Conditions, condition)
+		}, 5*time.Minute, 10*time.Second).Should(o.BeTrue())
+	})
+
+	g.It("Should function without any target pods", func(ctx context.Context) {
+		netConfigApply := applyconfigv1.Network(clusterConfig).WithSpec(
+			applyconfigv1.NetworkSpec().WithNetworkDiagnostics(
+				applyconfigv1.NetworkDiagnostics().
+					WithMode(configv1.NetworkDiagnosticsAll).
+					WithTargetPlacement(
+						applyconfigv1.NetworkDiagnosticsTargetPlacement().
+							WithNodeSelector(map[string]string{"alien": ""}),
+					),
+			),
+		)
+		_, err := oc.AdminConfigClient().ConfigV1().Networks().Apply(ctx, netConfigApply,
+			metav1.ApplyOptions{FieldManager: fieldManager, Force: true})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Eventually(func() bool {
+			pods, err := oc.AdminKubeClient().CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+				LabelSelector: "app=network-check-target"})
+			if err != nil {
+				framework.Logf("Error getting pods in %s namespace: %v", namespace, err)
+				return false
+			}
+			if len(pods.Items) != 0 {
+				framework.Logf("Target diagnostics pods found")
+				return false
+			}
+			cfg, err := oc.AdminConfigClient().ConfigV1().Networks().Get(ctx, clusterConfig, metav1.GetOptions{})
+			if err != nil {
+				framework.Logf("Error getting cluster config: %v", err)
+				return false
+			}
+			return meta.IsStatusConditionTrue(cfg.Status.Conditions, condition)
+		}, 5*time.Minute, 10*time.Second).Should(o.BeTrue())
+	})
+
+	g.It("Should set the condition to false if there are no nodes able to host the source pods", func(ctx context.Context) {
+		netConfigApply := applyconfigv1.Network(clusterConfig).WithSpec(
+			applyconfigv1.NetworkSpec().WithNetworkDiagnostics(
+				applyconfigv1.NetworkDiagnostics().
+					WithMode(configv1.NetworkDiagnosticsAll).
+					WithSourcePlacement(
+						applyconfigv1.NetworkDiagnosticsSourcePlacement().
+							WithNodeSelector(map[string]string{"alien": ""}),
+					),
+			),
+		)
+		_, err := oc.AdminConfigClient().ConfigV1().Networks().Apply(ctx, netConfigApply,
+			metav1.ApplyOptions{FieldManager: fieldManager, Force: true})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		o.Eventually(func() bool {
+			// Should not affect the Progressing condition of network.operator
+			oper, err := oc.AdminOperatorClient().OperatorV1().Networks().Get(ctx, clusterConfig, metav1.GetOptions{})
+			if err != nil {
+				framework.Logf("Error getting cluster operator: %v", err)
+				return false
+			}
+			for _, operCondition := range oper.Status.Conditions {
```

Comment:
Should we also check that CNO isn't progressing in the other tests where `meta.IsStatusConditionFalse`?

Reply:
I don't think it is required in other cases as nothing impacts pod deployment.